examples/js/shareable.js

```diff
@@ -0,0 +1,320 @@
+        currPagination = new Paginator(url, fetchedImages);
+        currPagination.processImgs(renderThumbnails, renderImages);
+        responseText.innerHTML = imageCount ? `${imageCount} image(s) fetched successfully from ${fetchedFrom.innerHTML}.` : 'No images found in the link provided...';
+      } else {
+        responseText.innerHTML = 'No images found in the link provided...';
+      }
+    })
+    .catch((error) => {
+      console.log(error);
+      responseText.innerHTML = 'Uh-oh! Something\'s not right with the link provided!';
+    })
+    .finally(() => {
+      bootstrap.Modal.getInstance(welcomeModal).hide();
+    });
+}
+
+welcomeModal.addEventListener('hidden.bs.modal', (event) => {
+  new bootstrap.Offcanvas(sidebar).show();
+  sidebarOpen = true;
+
+});
+
+restoreWelcomeModal.addEventListener('click', (event) => {
+  bootstrap.Modal.getInstance(welcomeModal).show();
+  input.value='';
+});
+
+mapToggle.addEventListener('click', (event) => {
+  new bootstrap.Offcanvas(sidebar).show();
+  sidebarOpen = true;
+});
+
+tileMap.addEventListener('click', (event) => {
+  if (sidebarOpen) {
+    bootstrap.Offcanvas.getInstance(sidebar).hide();
+  }
+});
+
+function getImageName(imageURL) {
+  const startIndex = imageURL.lastIndexOf('/') + 1;
+  const endIndex = imageURL.lastIndexOf('.');
+  const imageName = imageURL.substring(startIndex, endIndex);
+
+  return imageName;
+}
+// ----
+function extractJSONDownloadURL(url) {
+  const startIndex = url.lastIndexOf('=');
+  const jsonDownloadURL = url.slice(startIndex + 1);
+
+  return jsonDownloadURL;
+}
+
+function isJsonDetected(url) {
+  if (url.includes('?json=')) {
+    const startIndex = url.lastIndexOf('.');
+    const fileExtension = url.slice(startIndex + 1);
+
+    if (fileExtension === 'json') {
+      console.log('JSON found in map shareable link'); // left here purposely
+      return true;
+    }
+  }
+
+  return false;
+}
+
+//2. Connects to JSON file and fetches JSON data therein from remote source
+async function fetchRemoteJson(jsonDownloadURL) { //async function recreateMapFormJson(json) 
+  let index = 0;
+  const imgCollectionProps = [];
+
+  try {
+    const response = await axios.get(jsonDownloadURL);
+    if (response.data.images.length > 1) {
+      response.data.images.forEach((data) => {
+        imgCollectionProps[index] = data;
+        index++;
+      });
+      return {avg_cm_per_pixel: response.data.avg_cm_per_pixel, imgCollectionProps};
+    }
+    imgCollectionProps[index] = response.data.images;
+
+    return {avg_cm_per_pixel: response.data.avg_cm_per_pixel, imgCollectionProps};
+  } catch (err) {
+    console.log('err', err);
+  }
+}
+
+// 1. Performs preliminary checks and obtains map data (needed to reconstruct map in another function)
+// expects URL in this format: http://localhost:8081/examples/shareable.html?json=https://archive.org/download/segeotest/segeotest.json
+async function recreateMapDataFromJsonUrl(url) {  
```

Comment:
This function, I think, could go in the core library. We could move it in a separate PR, though if you prefer!

Reply:
Okay, so I will quickly fix this, no need for fresh PR. Many thanks!

Reply:
Hi @jywarren, still awaiting your reaction to the question. Thanks!

Reply:
Can we change `recreateMapDataFromJsonUrl()` so it accepts simply a direct URL to a JSON file, instead of `archive.html?json=https://example.com/map.json` format? Then we can move `recreateMapDataFromJsonUrl()` without moving `extractJSONDownloadURL`. 

Maybe `extractJSONDownloadURL` could be renamed `extractJsonFromUrlParams` to clarify, and `recreateMapDataFromJsonUrl` could be `recreateMapFromJsonUrl`? (just a little shorter and we are recreating the map itself, not just the data? Not a big deal but a good balance of descriptive and not extremely long :laugh:)

Reply:
And fantastic work on this!!!

Reply:
Okay, so meaning it's okay to move `isJsonDetected(url)` and `fetchRemoteJson(jsonDownloadURL)` along with `recreateMapDataFromJsonUrl()` right? Of course, the renaming will be done too.

Reply:
If recreateMapFromJsonUrl accepts only a direct URL and not a json= parameter, we wouldn't need to move isJsonDetected into the core library, since the url parameters are all managed outside in the application code, isn't that right? But I could be mixed up. But yes, fetchRemoteJson could move inside the core library too. Thank you!

Reply:
Okay, so that's correct. Mnay thanks!

Reply:
> Can we change `recreateMapDataFromJsonUrl()` so it accepts simply a direct URL to a JSON file, instead of `archive.html?json=https://example.com/map.json` format? Then we can move `recreateMapDataFromJsonUrl()` without moving `extractJSONDownloadURL`.
> 
> Maybe `extractJSONDownloadURL` could be renamed `extractJsonFromUrlParams` to clarify, and `recreateMapDataFromJsonUrl` could be `recreateMapFromJsonUrl`? (just a little shorter and we are recreating the map itself, not just the data? Not a big deal but a good balance of descriptive and not extremely long :laugh:)

Hi @jywarren, as per renaming `recreateMapDataFromJsonUrl` to `recreateMapFromJsonUrl`,  if you look at the function again, you will discover we are not re-creating map but reproducing data required to construct the map. Map is actually reconstructed in the user application (e.g., in our case archive.js) with the `createMap()`. This is where we create instance of the `distortableImageOverlay`. So it's my considered opinion that `recreateMapDataFromJsonUrl` (or the like) is a more appropriate name. *What do you think?*

Reply:
I see, and it makes sense. You could say "overlay" or "images" instead of "MapData"? But either is fine! Thank you!!

Reply:
Okay, thanks.